google/cloud/aiplatform/models.py

```diff
@@ -215,10 +244,548 @@ def upload(
             model_name=fields.id, project=fields.project, location=fields.location
         )
 
-    # TODO(b/169782716) add support for deployment when Endpoint class complete
-    def deploy(self):
-        raise NotImplementedError("Deployment not implemented.")
+    def deploy(
+        self,
+        endpoint_obj: Optional["Endpoint"] = None,
+        deployed_model_display_name: Optional[str] = None,
+        traffic_percentage: Optional[int] = 100,
+        machine_type: Optional[str] = None,
+        min_replica_count: Optional[int] = 0,
+        max_replica_count: Optional[int] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+        timeout: Optional[int] = None,
+    ) -> "Endpoint":
+        """
+        Deploys model to endpoint. Endpoint will be created if unspecified.
+
+        Args:
+            endpoint_obj ("Endpoint"):
+                Optional. Endpoint to deploy model to. If not specified, endpoint
+                display name will be model display name+'_endpoint'.
+            deployed_model_display_name (str):
+                Optional. The display name of the DeployedModel. If not provided
+                upon creation, the Model's display_name is used.
+            traffic_percentage (int):
+                Optional. Desired traffic to newly deployed model. Defaults to
+                100. Traffic of previously deployed models at the endpoint  will
+                be scaled down to accommodate new deployed model's traffic.
+            machine_type (str):
+                Optional. The type of machine. Following machine types are
+                supported:
```

Comment:
is this copied from GAPIC library's docstring?  this can quickly go out of date and become inaccurate.

Reply:
I'll remove the supported machine types. Does that sound good?

Reply:
If parts of the docstring are copied from GAPIC (which may be a very regular thing to do for this library) we might want to explore ways to automatically grab relevant pieces of the GAPIC docstring.

Reply:
I've been copying from the gapic docstring but it's not 1:1 as the mbsdk might change the behaviour prior to calling the gapic function.